src/retabled/core.cljc

```diff
@@ -155,18 +164,37 @@
      (render-screen-controls paging-controls))
    (render-header-fields controls SORT)])
 
+(defn resolve-click-to-filter
+  [controls entries]
+  (let [{:keys [click-to-filter valfn displayfn]} controls]
+    (cond
+      (and valfn (string? (valfn entries)))
+      (valfn entries)
+
+      (and displayfn (string? (displayfn entries)))
+      (displayfn entries)
+
+      (fn? click-to-filter)
```

Comment:
Click to filter needs to be the first option, so it takes precedence over the other two